File: app/src/main/java/com/devidea/timeleft/EntityItemInfo.java
package com.devidea.timeleft;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class EntityItemInfo {
    @PrimaryKey(autoGenerate = true)
    int id = 0;

    public String type;
    public String startValue;
    public String endValue;
    public String summery;
    public boolean autoUpdate;

    public EntityItemInfo(String type, String startValue, String endValue, String summery, boolean autoUpdate) {
        this.type = type;
        this.startValue = startValue;
        this.endValue = endValue;
        this.summery = summery;
        this.autoUpdate = autoUpdate;
    }

    public int getId() {
        return id;
    }

    public String getType() {
        return type;
    }

    public String getStartValue() {
        return startValue;
    }

    public String getEndValue() {
        return endValue;
    }

    public String getSummery() {
        return summery;
    }

    public boolean isAutoUpdate() {
        return autoUpdate;
    }
}

File: app/src/main/kotlin/com/devidea/timeleft/EntityItemInfo.kt
package com.devidea.timeleft

import com.devidea.timeleft.EntityItemInfo
import com.devidea.timeleft.MainActivity
import android.appwidget.AppWidgetProvider
import android.content.Intent
import android.appwidget.AppWidgetManager
import com.devidea.timeleft.AppDatabase
import com.devidea.timeleft.AppWidget
import android.app.PendingIntent
import android.app.AlarmManager
import android.widget.RemoteViews
import com.devidea.timeleft.R
import com.devidea.timeleft.AdapterItem
import com.devidea.timeleft.EntityWidgetInfo
import com.devidea.timeleft.DatabaseDao
import com.devidea.timeleft.InterfaceItem
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView
import com.devidea.timeleft.TopRecyclerView
import android.os.Bundle
import android.widget.TextView
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.PagerSnapHelper
import me.relex.circleindicator.CircleIndicator2
import android.content.DialogInterface
import com.devidea.timeleft.CreateTimeActivity
import com.devidea.timeleft.CreateMonthActivity
import android.widget.Toast
import android.os.Looper
import com.devidea.timeleft.BottomRecyclerView
import com.devidea.timeleft.ItemGenerate
import android.view.ViewGroup
import android.view.LayoutInflater
import android.animation.ObjectAnimator
import android.widget.ProgressBar
import android.app.Activity
import android.widget.Spinner
import android.widget.CheckBox
import android.widget.RadioGroup
import android.widget.ArrayAdapter
import android.widget.AdapterView
import android.util.SparseBooleanArray
import android.annotation.SuppressLint
import android.animation.ValueAnimator
import android.animation.ValueAnimator.AnimatorUpdateListener
import android.widget.EditText
import com.devidea.timeleft.ItemSave
import android.app.TimePickerDialog.OnTimeSetListener
import android.widget.TimePicker
import android.app.TimePickerDialog
import android.app.DatePickerDialog.OnDateSetListener
import android.widget.DatePicker
import android.app.DatePickerDialog
import androidx.room.*

@Entity
class EntityItemInfo constructor(
    var type: String,
    var startValue: String,
    var endValue: String,
    var summery: String?,
    var isAutoUpdate: Boolean
) {
    @PrimaryKey(autoGenerate = true)
    var id: Int = 0

}